app/components/primer/alpha/dialog.rb

```diff
@@ -0,0 +1,142 @@
+      }.freeze
+      HEIGHT_OPTIONS = HEIGHT_MAPPINGS.keys
+
+      # Optional button to open the dialog.
+      #
+      # @param system_arguments [Hash] The same arguments as <%= link_to_component(Primer::ButtonComponent) %>.
+      renders_one :show_button, lambda { |**system_arguments|
+        system_arguments[:classes] = class_names(
+          system_arguments[:classes]
+        )
+        system_arguments[:id] = "dialog-show-#{@system_arguments[:id]}"
+        system_arguments["data-show-dialog-id"] = @system_arguments[:id]
+        system_arguments[:data] = (system_arguments[:data] || {}).merge({ "show-dialog-id": @system_arguments[:id] })
+        Primer::ButtonComponent.new(**system_arguments)
+      }
+
+      # Header content.
+      #
+      # @param hide_divider [Boolean] If true the visual dividing line between the header and body will be hidden
+      # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+      renders_one :header, lambda { |hide_divider: false, **system_arguments|
+        Primer::Alpha::Dialog::Header.new(
+          id: @id,
+          title: @title,
+          subtitle: @subtitle,
+          hide_divider: hide_divider,
+          **system_arguments
+        )
+      }
+
+      # Required body content.
+      #
+      # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+      renders_one :body, "Body"
+
+      # Footer content.
+      #
+      # @param hide_divider [Boolean] If true the visual dividing line between the body and footer will be hidden
+      # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+      renders_one :footer, "Footer"
+
+      # @example Dialog with Cancel and Submit buttons
+      #   @description
+      #     An ID is provided which enables wiring of the open and close buttons to the dialog.
+      #   @code
+      #     <%= render(Primer::Alpha::Dialog.new(
+      #       title: "Dialog Example",
+      #       id: "my-dialog",
+      #     )) do |d| %>
+      #       <% d.show_button { "Show Dialog" } %>
+      #       <% d.body do %>
+      #         <p>Some content</p>
+      #       <% end %>
+      #       <% d.footer do %>
+      #         <%= render(Primer::ButtonComponent.new(data: { "close-dialog-id": "my-dialog" })) { "Cancel" } %>
+      #         <%= render(Primer::ButtonComponent.new(scheme: :primary)) { "Submit" } %>
+      #       <% end %>
+      #     <% end %>
+      # @param id [String] The id of the dialog.
+      # @param title [String] The title of the dialog.
+      # @param subtitle [String] The subtitle of the dialog. This will also set the `aria-describedby` attribute.
+      # @param width [Symbol] The width of the dialog. <%= one_of(Primer::Alpha::Dialog::WIDTH_OPTIONS) %>
+      # @param height [Symbol] The height of the dialog. <%= one_of(Primer::Alpha::Dialog::HEIGHT_OPTIONS) %>
+      # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+      def initialize(
```

Comment:
I love your approach of pairing down some of the logic so we can address it with Overlay. However, I want to point out that [at least one team](https://github.com/github/github/pull/232953/files#diff-69396c583a76bad41c8de7b68c6a0c58afe813644b58e624eebb4544ab49aeefR28) has a use case for the `narrow` bottom sheet position, so I'm not sure how we should handle that here. I recommended they add just the position they need right now. Would we leave the Experimental version in place while Overlay gets going, and then ship another version of Dialog later with all the responsive properties?

Reply:
Yeah I'd be happy to do this :+1: